src/main/java/io/github/hapjava/server/impl/HomekitRoot.java

```diff
@@ -83,11 +85,19 @@ public class HomekitRoot {
     this(
         label, DEFAULT_ACCESSORY_CATEGORY, webHandler, authInfo, new JmdnsHomekitAdvertiser(jmdns));
   }
+
+  public synchronized void batchUpdate() {
```

Comment:
Im still trying to understand the usage of nested batches.
When batchUpdate should be called?

If understand correctly after first call, madeChanges will false  and nestedBatches will be 1
And follow up call will increase nestedBatch but not change madeChanges

If the method should be used outside, please add the description

Reply:
I have zero intention for nestedBatches to be > 1, but it's just following how SubscriptionManager implements a parallel concept, so I just used it so it's safe for multiple calls.

Reply:
Got it. It was done the same way in subscriptionManager

Still not fully get the concept but it has nothing to do with this PR.